src/contracts/permissions/KeyRegistrar.sol

```diff
@@ -0,0 +1,405 @@
+
+        if (!keyInfo.isRegistered) {
+            // Create default values for an empty key
+            uint256[2] memory zeroArray = [uint256(0), uint256(0)];
+            return (BN254.G1Point(0, 0), BN254.G2Point(zeroArray, zeroArray));
+        }
+
+        (uint256 g1X, uint256 g1Y, uint256[2] memory g2X, uint256[2] memory g2Y) =
+            abi.decode(keyInfo.keyData, (uint256, uint256, uint256[2], uint256[2]));
+
+        return (BN254.G1Point(g1X, g1Y), BN254.G2Point(g2X, g2Y));
+    }
+
+    /// @inheritdoc IKeyRegistrar
+    function getECDSAKey(OperatorSet memory operatorSet, address operator) public view returns (bytes memory) {
+        // Validate operator set curve type
+        CurveType curveType = operatorSetCurveTypes[operatorSet.key()];
+        require(curveType == CurveType.ECDSA, InvalidCurveType());
+
+        KeyInfo memory keyInfo = operatorKeyInfo[operatorSet.key()][operator];
+        return keyInfo.keyData; // Returns the 20-byte address as bytes
+    }
+
+    /// @inheritdoc IKeyRegistrar
+    function getECDSAAddress(OperatorSet memory operatorSet, address operator) external view returns (address) {
+        return address(bytes20(getECDSAKey(operatorSet, operator)));
+    }
+
+    /// @inheritdoc IKeyRegistrar
+    function isKeyGloballyRegistered(
+        bytes32 keyHash
+    ) external view returns (bool) {
+        return globalKeyRegistry[keyHash];
+    }
+
+    /// @inheritdoc IKeyRegistrar
+    function getKeyHash(OperatorSet memory operatorSet, address operator) external view returns (bytes32) {
+        KeyInfo memory keyInfo = operatorKeyInfo[operatorSet.key()][operator];
+        CurveType curveType = operatorSetCurveTypes[operatorSet.key()];
+
+        if (!keyInfo.isRegistered) {
+            return bytes32(0);
+        }
+
+        return _getKeyHashForKeyData(keyInfo.keyData, curveType);
+    }
+
+    /**
+     * @notice Returns the message hash for ECDSA key registration
+     * @param operator The operator address
+     * @param operatorSet The operator set
+     * @param keyAddress The ECDSA key address
+     * @return The message hash for signing
+     */
+    function getECDSAKeyRegistrationMessageHash(
+        address operator,
+        OperatorSet memory operatorSet,
+        address keyAddress
+    ) external view returns (bytes32) {
+        bytes32 structHash = keccak256(
+            abi.encode(ECDSA_KEY_REGISTRATION_TYPEHASH, operator, operatorSet.avs, operatorSet.id, keyAddress)
+        );
+        return _calculateSignableDigest(structHash);
+    }
+
+    /**
+     * @notice Returns the message hash for BN254 key registration
+     * @param operator The operator address
+     * @param operatorSet The operator set
+     * @param keyData The BN254 key data
+     * @return The message hash for signing
+     */
+    function getBN254KeyRegistrationMessageHash(
+        address operator,
+        OperatorSet memory operatorSet,
+        bytes calldata keyData
```

Comment:
What is this keyData? Can you be more explicit as to what that is? I.e how to construct it or what it is?